### ISSUE TITLE
conda_package_handling.exceptions.ArchiveCreationError: Failed to clean up compressor

### ISSUE THREAD

**Issue by thomas-mueller** [conda/conda-package-handling#65]
I get the following error when building a conda package:
```
INFO:conda_build.build:Packaging dstk
INFO conda_build.build:build(1571): Packaging dstk
source tree in: /opt/conda/conda-bld/dstk_1598256196268/work
Resource usage statistics from building dstk:
   Process count: 1
   CPU time: Sys=0:00:00.0, User=0:00:00.0
   Memory: 1.3M
   Disk usage: 17.7K
   Time elapsed: 0:00:02.1
Packaging dstk
fatal: No names found, cannot describe anything.
INFO:conda_build.build:Packaging dstk-0+untagged.54.g6b6c70c2.dirty-py36_1
INFO conda_build.build:bundle_conda(891): Packaging dstk-0+untagged.54.g6b6c70c2.dirty-py36_1
Packaging dstk-0+untagged.54.g6b6c70c2.dirty-py36_1
fatal: No names found, cannot describe anything.
fatal: No names found, cannot describe anything.
fatal: No names found, cannot describe anything.
fatal: No names found, cannot describe anything.
compiling .pyc files...
number of files: 69
Fixing permissions
Traceback (most recent call last):
  File "/opt/conda/lib/python3.6/site-packages/conda_package_handling/api.py", line 82, in create
    out = SUPPORTED_EXTENSIONS[ext].create(prefix, file_list, out_fn, out_folder, **kw)
  File "/opt/conda/lib/python3.6/site-packages/conda_package_handling/tarball.py", line 145, in create
    '.tar.bz2', 'bzip2')
  File "/opt/conda/lib/python3.6/site-packages/conda_package_handling/tarball.py", line 78, in create_compressed_tarball
    _create_libarchive(fullpath, files, compression_filter, filter_opts)
  File "/opt/conda/lib/python3.6/site-packages/conda_package_handling/tarball.py", line 64, in _create_libarchive
    archive_utils.create_archive(fullpath, files, compression_filter, filter_opts)
  File "/opt/conda/lib/python3.6/site-packages/conda_package_handling/archive_utils.py", line 29, in create_archive
    raise ArchiveCreationError(message)
conda_package_handling.exceptions.ArchiveCreationError: Failed to clean up compressor
During handling of the above exception, another exception occurred:
Traceback (most recent call last):
  File "/opt/conda/bin/conda-build", line 11, in <module>
    sys.exit(main())
  File "/opt/conda/lib/python3.6/site-packages/conda_build/cli/main_build.py", line 469, in main
    execute(sys.argv[1:])
  File "/opt/conda/lib/python3.6/site-packages/conda_build/cli/main_build.py", line 460, in execute
    verify=args.verify, variants=args.variants)
  File "/opt/conda/lib/python3.6/site-packages/conda_build/api.py", line 209, in build
    notest=notest, need_source_download=need_source_download, variants=variants)
  File "/opt/conda/lib/python3.6/site-packages/conda_build/build.py", line 2344, in build_tree
    notest=notest,
  File "/opt/conda/lib/python3.6/site-packages/conda_build/build.py", line 1643, in build
    newly_built_packages = bundlers[pkg_type](output_d, m, env, stats)
  File "/opt/conda/lib/python3.6/site-packages/conda_build/build.py", line 1028, in bundle_conda
    basename + ext, out_folder=tmp)
  File "/opt/conda/lib/python3.6/site-packages/conda_package_handling/api.py", line 85, in create
    if _os.path.isfile(out):
UnboundLocalError: local variable 'out' referenced before assignment
```
The debug mode does not output more information. It looks like the package is successfully built but cannot be achived into a tar.bz2 file.

Actually, this problem only occurs when I run the `conda build` step in a Gitlab pipeline. It runs locally and during building a docker image (also within a Gitlab pipeline). I checked that I do exactly the same in all three cases and that versions of relevant packages are all the same. The output folder is created before the `conda build` step. I also checked that building other packages in Gitlab pipelines are possible.

There was a [similar issue](https://github.com/conda/conda-package-handling/issues/54) here, but there was no detailed solution postted before this issue was closed.

What else could I try? I am using conda build version 3.18.11 with python 3.6.

**Comment by dhirschfeld** [comment]
Running into this same issue.

**Comment by dhirschfeld** [comment]
I this should be classified as a bug since if something goes wrong the error should be caught and a meaningful/actionable error message should be shown to the user - not an `UnboundLocalError`.

**Comment by jezdez** [comment]
The UnboundLocalError at least was fixed in #77 and it raises the caught exception to the user.
